File: core/src/main/java/org/evergreen_ils/system/Organization.java
/*
 * Copyright (C) 2016 Kenneth H. Cox
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 */
package org.evergreen_ils.system;

import org.evergreen_ils.Api;
import org.evergreen_ils.barcodescan.Intents;
import org.opensrf.util.GatewayResponse;

import java.io.Serializable;
import java.util.Map;

public class Organization /*implements Serializable*/ {
    private static final String TAG = Organization.class.getSimpleName();

    public Integer level = null;
    public Integer id = null;
    public Integer parent_ou = null;
    public String name = null;
    public String shortname = null;
    public OrgType orgType = null;
    public String indentedDisplayPrefix = "";

    public Boolean opac_visible = null;
    public Boolean is_pickup_location = null; // null=not loaded

    public Organization() {
    }

    public String getTreeDisplayName() {
        return indentedDisplayPrefix + name;
    }

    public boolean isPickupLocation() {
        if (is_pickup_location != null) {
            return is_pickup_location;
        } else {
            return defaultIsPickupLocation();
        }
    }

    private boolean defaultIsPickupLocation() {
        if (orgType == null)
            return true;//should not happen
        return orgType.can_have_vols;
    }

    public boolean pickupLocationNeedsLoading() {
        if (is_pickup_location != null) {
            // already loaded
            return false;
        } else if (defaultIsPickupLocation() == false) {
            // pickup location can be disabled by network setting but not enabled
            return false;
        } else {
            return true;
        }
    }

    public void setIsPickupLocationFromGatewayResponse(GatewayResponse response) {
        try {
            Map<String, ?> resp_map = (Map<String, ?>) response.payload;
            Object o = resp_map.get(Api.SETTING_ORG_UNIT_NOT_PICKUP_LIB);
            if (o == null) {
                is_pickup_location = defaultIsPickupLocation();
            } else {
                Map<String, ?> resp_org_map = (Map<String, ?>)o;
                is_pickup_location = !Api.parseBoolean(resp_org_map.get("value"));
            }
            Log.d(TAG, name+" id "+id+" is_pickup_location="+is_pickup_location);
        } catch (Exception e) {
            Log.d(TAG, "caught", e);
            is_pickup_location = defaultIsPickupLocation();
        }
    }
}

File: core/src/main/java/net/kenstir/hemlock/data/models/Organization.kt
/*
 * Copyright (c) 2019 Kenneth H. Cox
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 */
package net.kenstir.hemlock.data.models

open class Organization(
    @JvmField val id: Int,
    @JvmField val level: Int,
    @JvmField val name: String,
    @JvmField val shortname: String,
    @JvmField val opacVisible: Boolean,
    @JvmField var parent: Int?
) {
    var email: String? = null
    var phone: String? = null
    var eresourcesUrl: String? = null
    var eventsURL: String? = null
    var infoURL: String? = null
    var meetingRoomsUrl: String? = null
    var museumPassesUrl: String? = null
    open val isConsortium: Boolean = false
    var isPaymentAllowed: Boolean = false
    open val isPickupLocation: Boolean = true
    var settingsLoaded = false

    // display fields
    var indentedDisplayPrefix = ""
    val spinnerLabel: String
        get() = indentedDisplayPrefix + name

    open fun getAddress(separator: String = " "): String {
        return ""
    }
}
